linting: type checking failures with dulwich 0.20.36

There have been some [type checking related changes](https://github.com/jelmer/dulwich/compare/upstream/0.20.35...dulwich-0.20.36) in the latest dulwich release ([0.20.36](https://github.com/jelmer/dulwich/releases/tag/dulwich-0.20.36)) which break mypy type checking:

```console
nox > python -m mypy
scmrepo/git/backend/dulwich/__init__.py: note: In member "fetch_refspecs" of class "DulwichBackend":
scmrepo/git/backend/dulwich/__init__.py:630:33: error: Argument
"determine_wants" to "fetch" of "GitClient" has incompatible type
"Callable[[Any], Any]"; expected
"Optional[Callable[[Dict[bytes, bytes], Optional[int]], List[bytes]]]" 
[arg-type]
                    determine_wants=determine_wants,
                                    ^
Found 1 error in 1 file (checked 26 source files)
nox > Command python -m mypy failed with exit code 1
nox > Session lint failed.
```

--- a/scmrepo/git/backend/dulwich/__init__.py
+++ b/scmrepo/git/backend/dulwich/__init__.py
@@ -581,7 +581,7 @@ class DulwichBackend(BaseGitBackend):  # pylint:disable=abstract-method
         self,
         url: str,
         refspecs: Union[str, Iterable[str]],
-        force: Optional[bool] = False,
+        force: bool = False,
         on_diverged: Optional[Callable[[str, str], bool]] = None,
         progress: Callable[["GitProgressEvent"], None] = None,
         **kwargs,
@@ -597,12 +597,17 @@ class DulwichBackend(BaseGitBackend):  # pylint:disable=abstract-method
 
         fetch_refs = []
 
-        def determine_wants(remote_refs):
+        def determine_wants(
+            remote_refs: Dict[bytes, bytes],
+            depth: Optional[int] = None,  # pylint: disable=unused-argument
+        ) -> List[bytes]:
             fetch_refs.extend(
                 parse_reftuples(
                     remote_refs,
                     self.repo.refs,
-                    refspecs,
+                    os.fsencode(refspecs)
+                    if isinstance(refspecs, str)
+                    else [os.fsencode(refspec) for refspec in refspecs],
                     force=force,
                 )
             )

--- a/setup.cfg
+++ b/setup.cfg
@@ -25,7 +25,7 @@ zip_safe = False
 packages = find:
 install_requires=
     gitpython>3
-    dulwich>=0.20.34
+    dulwich>=0.20.37
     pygit2>=1.7.2
     pygtrie>=2.3.2
     fsspec>=2021.7.0

